### PR TITLE
F5 Big IP Cookie Decoder

### PR DESCRIPTION
**Pull request by m-q-t**
Hi team,

Please find attached the F5 Big IP Cookie Decoder task.

**Behavior**:
The way the task works is by sending 10 requests to the application and decodes the cookie of each request. The amount of requests was adopted from the MSF Module and is done so because each time the cookie is decoded a different Internal IP Address can be returned. The caveat is that if the task doesn't successfully decode a cookie by the second request, it ceases firing off requests in order to suppress the noisiness. 

In the scenario where a F5 Big IP Cookie is successfully decoded; it will clear out the duplicate IP Addresses and create an issue which will display all the leaked pool names, internal Ip Addresses and ports. A new entity will be created per IP Address as well.

image

image

Best regards,
Maxim

### CODE REVIEW

File: lib/tasks/f5_bigip_cookie_decoder.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          ip_address = Regexp.last_match(1).to_i(16)
+          port = Regexp.last_match(2).to_i
+          ip_address = addr_itoa(ip_address, true)
+          port = change_endianness(port, 2)
+        when cookie_value.downcase =~ /rd\d+o([a-f0-9]{32})o(\d{1,5})/
+          ip_address = Regexp.last_match(1).to_i(16)
+          port = Regexp.last_match(2).to_i
+          ip_address = addr_itoa(ip_address, true)
+        else
+          ip_address = nil
```
Comment (shpendk) [inline lib/tasks/f5_bigip_cookie_decoder.rb:73]: If we land in the else branch, we should note that this is not an issue and not raise the issue

**Review by shpendk — CHANGES_REQUESTED** [review]
See the inline comment

**Comment by m-q-t** [comment]
@shpendk 

Thanks for looking into this + catching the bug. Should be fixed with the latest commit.

Best, 
Maxim

**Comment by jcran** [comment]
Great work on this, merging.

**Pull request merged**